### DISCUSSION TITLE
Problems with installation of OMNeT++ federate

### DISCUSSION THREAD

**Discussion by abdullah-traboulsi** [eclipse/mosaic#41] [Q&A]
I have tried installing Omnetpp federate with Mosaic provided from the link
https://www.eclipse.org/mosaic/docs/simulators/network_simulator_omnetpp/

But I received the following error when trying to install it
I would like to ask how to resolve it, thank you

I believe the error was due to protoc version even though the version of 3.6.1

protoc --version
libprotoc 3.6.1

```
This file was generated by an older version of protoc which is incompatible with your Protocol Buffer headers. Please regenerate this file with a newer version of protoc

src/util/ClientServerChannelMessages.pb.h:17:2: error: #error This file was generated by an older version of protoc which is
   17 | #error This file was generated by an older version of protoc which is
      |  ^~~~~
src/util/ClientServerChannelMessages.pb.h:18:2: error: #error incompatible with your Protocol Buffer headers. Please
   18 | #error incompatible with your Protocol Buffer headers.  Please
      |  ^~~~~
src/util/ClientServerChannelMessages.pb.h:19:2: error: #error regenerate this file with a newer version of protoc.
   19 | #error regenerate this file with a newer version of protoc.
      |  ^~~~~
In file included from src/util/ClientServerChannelMessages.pb.h:27,
                 from src/util/ClientServerChannel.h:27,
                 from src/mgmt/MosaicEventScheduler.h:28,
                 from src/mgmt/MosaicEventScheduler.cc:23:
/usr/include/google/protobuf/inlined_string_field.h:56:22: error: expected ‘;’ at end of member declaration
   56 |   InlinedStringField()

src/util/ClientServerChannelMessages.pb.h: In member function ‘void ClientServerChannelSpace::SendMessageMessage::clear_length()’:
src/util/ClientServerChannelMessages.pb.h:3473:13: error: ‘GOOGLE_ULONGLONG’ was not declared in this scope; did you mean ‘GOOGLE_DLOG’?
 3473 |   length_ = GOOGLE_ULONGLONG(0);
      |             ^~~~~~~~~~~~~~~~
      |             GOOGLE_DLOG
make[1]: *** [omnetpp-federate-LIBRARY.make:176: obj/Debug/omnetpp-federate-LIBRARY/MosaicEventScheduler.o] Error 1
make: *** [Makefile:29: omnetpp-federate-LIBRARY] Error 2
```

**Comment by kschrab** [comment]
We would need a bit more infos of your system for this. Somehow it seems that you have installed multiple versions of protobuf. The version of the protobuf compiler (protoc) must match with the protobuf library.

**Reply by abdullah-traboulsi** [reply]
I am not sure what version of protoc do I need to have to successfully build the federates?
How can I check the different versions in my system?
Can I point the federate to use a specific version of protoc during build?

I am using a Ubuntu 20 version

protoc --version
libprotoc 3.6.1

Command: pip show protobuf
Name: protobuf
Version: 3.14.0
Summary: Protocol Buffers
Home-page: https://developers.google.com/protocol-buffers/
Author: None
Author-email: None
License: 3-Clause BSD License
Location: /home/abdullah/.pyenv/versions/3.6.7/lib/python3.6/site-packages
Requires: six
Required-by:
